contracts/vaults/RibbonDeltaVault.sol

```diff
@@ -148,6 +151,48 @@ contract RibbonDeltaVault is RibbonVault, OptionsDeltaVaultStorage {
         optionAllocationPct = newOptionAllocationPct;
     }
 
+    /**
+     * @notice Withdraws the assets on the vault using the outstanding `DepositReceipt.amount`
+     * @param amount is the amount to withdraw
+     */
+    function withdraw(uint256 amount) external nonReentrant {
+        require(amount > 0, "!amount");
+
+        uint256 amountLeftForWithdrawal = _withdrawFromNewDeposit(amount);
+
+        uint16 currentRound = vaultState.round;
+
+        // If we need to withdraw beyond current round deposit, we will need to
+        // use round pps as it had exposure to at least one week's worth of options
+        if (amountLeftForWithdrawal > 0) {
+            uint256 sharesLeft =
+                ShareMath.underlyingToShares(
+                    amountLeftForWithdrawal,
+                    roundPricePerShare[currentRound],
+                    vaultParams.decimals
+                );
+
+            (uint256 heldByAccount, uint256 heldByVault) =
+                shareBalances(msg.sender);
+
+            uint256 totalShares = heldByAccount.add(heldByVault);
+
+            require(sharesLeft <= totalShares, "Insufficient balance");
+
+            // Burn both redeemed shares and custodied shares of user
+            if (heldByAccount > 0) {
+                _burn(msg.sender, heldByAccount);
+            }
+            if (heldByVault > 0) {
+                _burn(address(this), heldByVault);
+            }
```

Comment:
Based on ChainSafe's comments, we should avoid using this pattern of burning from both the vault and account on Theta Vault. I tend to agree.

One way to refactor this is to force a maxRedeem to happen before we burn, so we burn the shares from the user's account instead